### ISSUE TITLE
Unable to create new instance of custom type in Dictionary

### ISSUE THREAD

**Issue by wizcas** [vexe/VFW#70]
I'm trying to use a dictionary with my own defined type (say, `Dictionary<string, SceneInfo>`), but it seems impossible to properly create a new instance of SceneInfo and serialize it with Vexe editor. #The value member keeps null, no matter how many items I click the type name in SelectionWindow.#

I debugged the code and found that the new instance of `SceneInfo` was correctly created and set to the underlying `EditorMember` within `TryCreateInstance`. The problem seems to be that it failed to write back to the dictionary.

I was totally confused by the mult-level nested EditorMember things, and now hopeless to get it done. I believe it's a bug. Could you take a look and make it right? Thanks.

**Comment by wizcas** [comment]
And by the way, the `SceneInfo` in the example is not a `MonoBehaviour`/`BetterBehaviour`

**Comment by vexe** [comment]
Is it a struct or class? Have you tried 'right clicking' on the object selection circle?

**Comment by vexe** [comment]
Did that solve your problem @wizcas?

**Comment by bastrakov-sergei** [comment]
Hello,
I have same problem with Dictionary. (Dictionary<string, CustomType>)
If CustomType is not abstract, 'right click' is working. But selection type in SelectionWindow does not result.
If CustomType is abstract, 'right click' does not work, as expected. In SelectionWindow I see all CustomType's child, but selection does not work.

**Comment by vexe** [comment]
Right clicking doesn't work on an abstract class because, well it's abstract, you can't instantiate an instance of it. I'll add testing the selection to my notes.

**Comment by bastrakov-sergei** [comment]
Yes, I know why abstract class can not be instatntiate, but it seems I don't know English =)
Now I use List<CustomType>, and "Key" from dictionary moved in CustomType as property. It not exactly as originally planned, but it solves my problem.
If I find anything else, I'll tell you.
